Linking against LibreSSL fails with CMake when using static libs

Trying to use LibreSSL in a CMake project, and building with `BUILD_SHARED_LIBS=OFF` results in errors at the linking stage of my program. The linker complains about missing `pthread` symbols:

```
[100%] Linking C executable myssltest
vendor/libressl/crypto/libcrypto.a(err.c.o): In function `ERR_load_ERR_strings':
err.c:(.text+0xb4e): undefined reference to `pthread_once'
vendor/libressl/crypto/libcrypto.a(crypto_init.c.o): In function `OPENSSL_init_crypto':
crypto_init.c:(.text+0x51): undefined reference to `pthread_once'
vendor/libressl/crypto/libcrypto.a(conf_sap.c.o): In function `OpenSSL_config':
conf_sap.c:(.text+0xbc): undefined reference to `pthread_once'
vendor/libressl/crypto/libcrypto.a(conf_sap.c.o): In function `OpenSSL_no_config':
conf_sap.c:(.text+0xf3): undefined reference to `pthread_once'
vendor/libressl/crypto/libcrypto.a(eng_all.c.o): In function `ENGINE_load_builtin_engines':
eng_all.c:(.text+0x30): undefined reference to `pthread_once'
vendor/libressl/crypto/libcrypto.a(err_all.c.o):err_all.c:(.text+0x9f): more undefined references to `pthread_once' follow
collect2: error: ld returned 1 exit status
CMakeFiles/myssltest.dir/build.make:88: recipe for target 'myssltest' failed
make[2]: *** [myssltest] Error 1
CMakeFiles/Makefile2:1239: recipe for target 'CMakeFiles/myssltest.dir/all' failed
make[1]: *** [CMakeFiles/myssltest.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2
```

Likely the problem is that the LibreSSL libraries do not declare their dependencies to `libpthread` (correctly captured in the `PLATFORM_LIBS` variable) unless they are build as shared libraries:
https://github.com/libressl-portable/portable/blob/14dd93c37a2a38dee7b8d2e0b823b0e0cc21e8d4/ssl/CMakeLists.txt#L64-L75
Similar `if` statements exist for `crypto` and `tls`.

--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -983,18 +983,16 @@ target_include_directories(crypto
 	PUBLIC
 		../include)
 
-if (BUILD_SHARED_LIBS)
-	export_symbol(crypto ${CMAKE_CURRENT_BINARY_DIR}/crypto_p.sym)
-	target_link_libraries(crypto ${PLATFORM_LIBS})
-	if (WIN32)
-		set(CRYPTO_POSTFIX -${CRYPTO_MAJOR_VERSION})
-	endif()
-	set_target_properties(crypto PROPERTIES
-		OUTPUT_NAME crypto${CRYPTO_POSTFIX}
-		ARCHIVE_OUTPUT_NAME crypto${CRYPTO_POSTFIX})
-	set_target_properties(crypto PROPERTIES VERSION
-		${CRYPTO_VERSION} SOVERSION ${CRYPTO_MAJOR_VERSION})
+export_symbol(crypto ${CMAKE_CURRENT_BINARY_DIR}/crypto_p.sym)
+target_link_libraries(crypto ${PLATFORM_LIBS})
+if (WIN32)
+	set(CRYPTO_POSTFIX -${CRYPTO_MAJOR_VERSION})
 endif()
+set_target_properties(crypto PROPERTIES
+	OUTPUT_NAME crypto${CRYPTO_POSTFIX}
+	ARCHIVE_OUTPUT_NAME crypto${CRYPTO_POSTFIX})
+set_target_properties(crypto PROPERTIES VERSION
+	${CRYPTO_VERSION} SOVERSION ${CRYPTO_MAJOR_VERSION})
 
 if(ENABLE_LIBRESSL_INSTALL)
 	install(

--- a/ssl/CMakeLists.txt
+++ b/ssl/CMakeLists.txt
@@ -61,18 +61,16 @@ target_include_directories(ssl
 	PUBLIC
 		../include)
 
-if (BUILD_SHARED_LIBS)
-	export_symbol(ssl ${CMAKE_CURRENT_SOURCE_DIR}/ssl.sym)
-	target_link_libraries(ssl crypto ${PLATFORM_LIBS})
-	if (WIN32)
-		set(SSL_POSTFIX -${SSL_MAJOR_VERSION})
-	endif()
-	set_target_properties(ssl PROPERTIES
-		OUTPUT_NAME ssl${SSL_POSTFIX}
-		ARCHIVE_OUTPUT_NAME ssl${SSL_POSTFIX})
-	set_target_properties(ssl PROPERTIES VERSION ${SSL_VERSION}
-		SOVERSION ${SSL_MAJOR_VERSION})
+export_symbol(ssl ${CMAKE_CURRENT_SOURCE_DIR}/ssl.sym)
+target_link_libraries(ssl crypto ${PLATFORM_LIBS})
+if (WIN32)
+	set(SSL_POSTFIX -${SSL_MAJOR_VERSION})
 endif()
+set_target_properties(ssl PROPERTIES
+	OUTPUT_NAME ssl${SSL_POSTFIX}
+	ARCHIVE_OUTPUT_NAME ssl${SSL_POSTFIX})
+set_target_properties(ssl PROPERTIES VERSION ${SSL_VERSION}
+	SOVERSION ${SSL_MAJOR_VERSION})
 
 if(ENABLE_LIBRESSL_INSTALL)
 	install(

--- a/tls/CMakeLists.txt
+++ b/tls/CMakeLists.txt
@@ -37,18 +37,16 @@ target_include_directories(tls
 	PUBLIC
 		../include)
 
-if (BUILD_SHARED_LIBS)
-	export_symbol(tls ${CMAKE_CURRENT_SOURCE_DIR}/tls.sym)
-	target_link_libraries(tls ssl crypto ${PLATFORM_LIBS})
-	if (WIN32)
-		set(TLS_POSTFIX -${TLS_MAJOR_VERSION})
-	endif()
-	set_target_properties(tls PROPERTIES
-		OUTPUT_NAME tls${TLS_POSTFIX}
-		ARCHIVE_OUTPUT_NAME tls${TLS_POSTFIX})
-	set_target_properties(tls PROPERTIES VERSION ${TLS_VERSION}
-		SOVERSION ${TLS_MAJOR_VERSION})
+export_symbol(tls ${CMAKE_CURRENT_SOURCE_DIR}/tls.sym)
+target_link_libraries(tls ssl crypto ${PLATFORM_LIBS})
+if (WIN32)
+	set(TLS_POSTFIX -${TLS_MAJOR_VERSION})
 endif()
+set_target_properties(tls PROPERTIES
+	OUTPUT_NAME tls${TLS_POSTFIX}
+	ARCHIVE_OUTPUT_NAME tls${TLS_POSTFIX})
+set_target_properties(tls PROPERTIES VERSION ${TLS_VERSION}
+	SOVERSION ${TLS_MAJOR_VERSION})
 
 if(ENABLE_LIBRESSL_INSTALL)
 	install(

